### PR TITLE
Update regression tests to run!

### LINKED ISSUE #229: Need ability for ncdata_check routines to end model run
### What is the feature/what would you like to discuss?

As a first step to the eventual CCPP physics testbed, it would be beneficial if one could add a new namelist entry that would allow the `ncdata_check` snapshot file-checking system to end the model simulation if differences are found.

In particular, this feature would allow one to setup basic smoke tests to automatically check whether a physics scheme/suite was preserving answers, without having to do any sort of complicated parsing of the generated log files after the model test run has finished.

### Is there anyone in particular you want to be part of this conversation?

_No response_

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

Any Software Engineer can do this

### PR DESCRIPTION
**Pull request by peverwhee**
Tag name (required for release branches): sima0_00_001
Originator(s): peverwhee

Description (include the issue title, and the keyword ['closes', 'fixes', 'resolves'] followed by the issue number):
closes #229 (Need ability for ncdata_check routines to end model run)
addresses #230 (Need CIME smoke tests for FPHYStest compset and various physics suites)

**Updates the regression tests to actually run and compare against previous baselines.**

Describe any changes made to build system: 
- added new compset FCAM7 that exercises the cam7 physics suites
- updated config_pes.xml for ne3 grid

Describe any changes made to the namelist:
- updated namelist defaults for ne3 grid (though these are currently unused; using analytic_ic until the SE dycore is updated)

List any changes to the defaults for the input datasets (e.g. boundary datasets): none

List all files MOVED and why:
cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_mpas_derecho_nooutput/user_nl_cam
cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_mpas_derecho_nooutput/shell_commands
cime_config/testdefs/testmods_dirs/cam/outfrq_held_suarez_derecho_nooutput/shell_commands
cime_config/testdefs/testmods_dirs/cam/outfrq_held_suarez_derecho_nooutput/user_nl_cam
cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_derecho_nooutput/shell_commands
cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_derecho_nooutput/user_nl_cam
cime_config/testdefs/testmods_dirs/cam/outfrq_tj2016_derecho_nooutput/user_nl_cam
cime_config/testdefs/testmods_dirs/cam/outfrq_tj2016_derecho_nooutput/shell_commands
- removed "nooutput" from directory name

cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_se_cslam_derecho_nooutput/user_nl_cam
cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_se_cslam_derecho_nooutput/shell_commands
- moved to outfrq_se_sclam/*

List all files added and what they do:
A cime_config/config_archive.xml
- brought over from CAM so tests will run

List all existing files that have been modified, and describe the changes: 
(Helpful git command: `git diff --name-status development...<your_branch_name>`)
M cime_config/config_component.xml
- added FCAM7 configuration

M cime_config/config_compset.xml
- added FCAM7 compset

M cime_config/config_pes.xml
- modified for ne3 grid

M cime_config/namelist_definition_cam.xml
- added ne3 grid defaults

M cime_config/testdefs/testlist_cam.xml
- added FHS94 and FTJ16 SE CSLAM tests
- modified existing FKESSLER SE CSLAM test to use ne5

M src/dynamics/tests/namelist_definition_analy_ic.xml
- use held-suarez initial conditions by default (not "none")

M src/physics/utils/phys_comp.F90
M src/physics/utils/physics_data.F90
- add namelist variable to endrun on diff found during ncdata_check

M test/system/archive_baseline.sh
- update to add a symlink to the ${baseline_dir}/latest_${CAM_FC} directory for easier test_driver usage
- add flag to turn off symlink addition

M test/system/test_driver.sh
- use ${baseline_dir}/latest_${CAM_FC} by default
- add flag to turn off default usage (no baseline comparison)

If there are new failures (compared to the `test/existing-test-failures.txt` file),
have them OK'd by the gatekeeper, note them here, and add them to the file.
If there are baseline differences, include the test and the reason for the
diff. What is the nature of the change? Roundoff?

derecho/intel/aux_sima:
SMS_Ln2.mpasa480_mpasa480.FKESSLER.derecho_intel.cam-outfrq_kessler_mpas_derecho
- will fail until MPAS is fully integrated

derecho/gnu/aux_sima:
SMS_Ln2.mpasa480_mpasa480.FKESSLER.derecho_gnu.cam-outfrq_kessler_mpas_derecho
- will fail until MPAS is fully integrated

If this changes climate describe any run(s) done to evaluate the new
climate in enough detail that it(they) could be reproduced: n/a

CAM-SIMA date used for the baseline comparison tests if different than latest: first baselines!

### CODE REVIEW

**Review by nusbaume — CHANGES_REQUESTED** [review]
Thanks for getting the tests going!  I do have some change requests (which is mostly me backpedaling from previous things I asked you to do, sorry!), but hopefully it won't be too hard to implement.  Of course please let me know if that is not the case.  Thanks again!

File: cime_config/testdefs/testmods_dirs/cam/outfrq_held_suarez_derecho/user_nl_cam
Comment (peverwhee) [reply cime_config/testdefs/testmods_dirs/cam/outfrq_held_suarez_derecho/user_nl_cam:5]: Currently, "ZM" is geopotential height (wrt surface) in CAM-SIMA - should this be "Z3" instead?

Comment (nusbaume) [reply cime_config/testdefs/testmods_dirs/cam/outfrq_held_suarez_derecho/user_nl_cam:5]: Whoops, sorry, I meant `ZM`, I just knew it as `Z3` from CAM (but I never liked the name because I have no idea what the "3" means).  Thanks!

File: cime_config/config_component.xml
Comment (peverwhee) [reply cime_config/config_component.xml]: moved & added a new outfrq_se_cslam_analy_ic test case directory

File: cime_config/testdefs/testlist_cam.xml
Comment (peverwhee) [reply cime_config/testdefs/testlist_cam.xml]: updated!

File: cime_config/testdefs/testlist_cam.xml
Comment (peverwhee) [reply cime_config/testdefs/testlist_cam.xml:42]: updated!

File: cime_config/testdefs/testlist_cam.xml
Comment (peverwhee) [reply cime_config/testdefs/testlist_cam.xml]: updated!

File: cime_config/testdefs/testlist_cam.xml
Comment (peverwhee) [reply cime_config/testdefs/testlist_cam.xml]: updated!

File: cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_mpas_derecho/user_nl_cam
Comment (peverwhee) [reply cime_config/testdefs/testmods_dirs/cam/outfrq_kessler_mpas_derecho/user_nl_cam]: added fields!

File: cime_config/testdefs/testmods_dirs/cam/outfrq_se_cslam/user_nl_cam
Comment (peverwhee) [reply cime_config/testdefs/testmods_dirs/cam/outfrq_se_cslam/user_nl_cam]: added fields!

File: cime_config/testdefs/testmods_dirs/cam/outfrq_held_suarez_derecho/user_nl_cam
Comment (peverwhee) [reply cime_config/testdefs/testmods_dirs/cam/outfrq_held_suarez_derecho/user_nl_cam]: changed!

File: src/dynamics/tests/namelist_definition_analy_ic.xml
Comment (peverwhee) [reply src/dynamics/tests/namelist_definition_analy_ic.xml]: switched back to none and updated the cam7 user_nl_cam file

File: test/system/archive_baseline.sh
Comment (peverwhee) [reply test/system/archive_baseline.sh]: fixed!

File: test/system/archive_baseline.sh
Comment (peverwhee) [reply test/system/archive_baseline.sh]: updated!

File: test/system/archive_baseline.sh
Comment (peverwhee) [reply test/system/archive_baseline.sh]: no reason i can think of - changed!

File: test/system/archive_baseline.sh
Comment (peverwhee) [reply test/system/archive_baseline.sh]: thanks! that does indeed work! changed!

**Comment by peverwhee** [comment]
@nusbaume thanks for your review. I also added the NL option to endrun on ncdata_check failures (and added it to the kessler test, which is currently the only one that has very small diffs) - if you could take a look at that, that'd be great!

**Review by nusbaume — CHANGES_REQUESTED** [review]
Almost there!   Just two last requests.

File: cime_config/testdefs/testlist_cam.xml
Comment (peverwhee) [reply cime_config/testdefs/testlist_cam.xml]: it was! thanks! fixed now.

File: src/data/write_init_files.py
Comment (peverwhee) [reply src/data/write_init_files.py]: Good idea - moved!

**Review by nusbaume — APPROVED** [review]
Looks great to me now.  Thanks again!

**Comment by nusbaume** [comment]
Also forgot to mention that you can close issue #229 once this PR is merged.  It might also be worth "addressing" issue #230 as well (we can close that one once we have all of snapshot checks working).  Thanks!

**Review by mwaxmonsky — APPROVED** [review]
Looks great @peverwhee, thanks for getting this added!

One minor whitespace fix I think might be needed but nothing worth holding up this PR.

File: test/system/test_driver.sh
Comment (peverwhee) [reply test/system/test_driver.sh:137]: Can I ask why this whitespace change? I'm not a bash expert, but it does match the rest of the arguments

Comment (peverwhee) [reply test/system/test_driver.sh]: turns out it was a "tab" whitespace - fixed!

**Pull request merged**
